Containerized Sorters problem

I tried to use example code from the link https://spikeinterface.readthedocs.io/en/latest/containerized_sorters.html#running-sorters-in-container-docker-singularity. It output the assertion error `must give a list of numpy array`

The environment is ubuntu 20.04, python 3.9, and spikeinterface 0.96.1.

The code for testing is
```
import spikeinterface.extractors as se
import spikeinterface.sorters as ss

test_recording, _ = se.toy_example(
    duration=30,
    seed=0,
    num_channels=64,
    num_segments=1
)

sorting = ss.run_waveclus(
    recording=test_recording,
    output_folder="waveclus",
    singularity_image=True)
```
 (I want to use waveclus sorter, and the kilosort3 sorter gives same assertion error)

The exception info:
```
SpikeSortingError: Spike sorting in singularity failed with the following error:
Traceback (most recent call last):
  File "/home/xxx/temp/in_container_sorter_script.py", line 9, in <module>
    recording = load_extractor('/home/cuilab/temp/in_container_recording.json')
  File "/home/xxx/temp/in_container_python_base/lib/python3.8/site-packages/spikeinterface/core/base.py", line 906, in load_extractor
    return BaseExtractor.load(file_or_folder_or_dict, base_folder=base_folder)
  File "/home/xxx/temp/in_container_python_base/lib/python3.8/site-packages/spikeinterface/core/base.py", line 545, in load
    extractor = BaseExtractor.from_dict(d, base_folder=base_folder)
  File "/home/xxx/temp/in_container_python_base/lib/python3.8/site-packages/spikeinterface/core/base.py", line 372, in from_dict
    extractor = _load_extractor_from_dict(d)
  File "/home/xxx/temp/in_container_python_base/lib/python3.8/site-packages/spikeinterface/core/base.py", line 849, in _load_extractor_from_dict
    extractor = cls(**kwargs)
  File "/home/xxx/temp/in_container_python_base/lib/python3.8/site-packages/spikeinterface/core/numpyextractors.py", line 30, in __init__
    assert all(isinstance(e, np.ndarray) for e in traces_list), 'must give a list of numpy array'
AssertionError: must give a list of numpy array
```
Why does this problem arise? waveclus sorter only uses CPU, so CUDA should not be the reason.

--- a/doc/containerized_sorters.rst
+++ b/doc/containerized_sorters.rst
@@ -54,6 +54,7 @@ The following code creates a test recording and runs a containerized spike sorte
         num_channels=64,
         num_segments=1
     )
+    test_recording = test_recording.save(folder="test-docker-folder")
 
     sorting = ss.run_kilosort3(
         recording=test_recording,
@@ -79,6 +80,13 @@ To use a specific image, set either ``docker_image`` or ``singularity_image`` to
         recording=test_recording,
         output_folder="kilosort3",
         singularity_image="spikeinterface/kilosort3-compiled-base:0.1.0")
+
+
+**NOTE:** the :code:`toy_example()` returns in-memory objects, which are not bound to a file on disk. 
+In order to run spike sorting in a container, the recording object MUST be persistent on disk, so that the 
+container can reload it. The :code:`save()` function makes the recording persistent on disk, by saving the in-memory 
+:code:`test_recording` object to a binary file in the :code:`test-docker-folder` folder.
+
 
 Contributing
 ------------

